File: gaia-sdk-python/gaia_sdk/graphql/request/type/Retrieval.py

from gaia_sdk.graphql.request.type.Experience import Experience
from gaia_sdk.graphql.request.type.Knowledge import Knowledge

from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField
from gaia_sdk.graphql.request.enumeration.EdgeType import EdgeType


class Retrieval(list):

    """
    Container element which collects all information static data
    """
    def knowledge(self, config: Callable[['Knowledge'], None]):
        def callback(registry: VariableRegistry):
            entity = Knowledge()
            config(entity)
            return "knowledge {" + entity.render(registry) + "}"
        self.append(callback)

    """
    Container element which collects all information about runtime data
    """
    def experience(self, config: Callable[['Experience'], None]):
        def callback(registry: VariableRegistry):
            entity = Experience()
            config(entity)
            return "experience {" + entity.render(registry) + "}"
        self.append(callback)

    def render(self, registry: VariableRegistry):
        return " ".join(map(lambda e: e(registry), self))

File: gaia-sdk-javascript/src/graphql/request/type/Retrieval.ts

import {Experience} from "./Experience";
import {Knowledge} from "./Knowledge";

import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

export class Retrieval extends Array<(_:VariableRegistry) => string> {
public _typeName = "Retrieval";
    /**
     * Container element which collects all information static data
     */
    public knowledge = (config: (_:Knowledge) => void) => this.push((registry) => {
        const entity = new Knowledge();
        config(entity);
        return "knowledge { " + entity.render(registry) + " }";
    });

    /**
     * Container element which collects all information about runtime data
     */
    public experience = (config: (_:Experience) => void) => this.push((registry) => {
        const entity = new Experience();
        config(entity);
        return "experience { " + entity.render(registry) + " }";
    });

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

